src/Kernel/client/plotting.ts

```diff
@@ -27,6 +27,276 @@ export interface DisplayableState {
+                    }),
+                    backgroundColor: "#5390d9",
+                    borderColor: "#5390d9",
+                }
+            ],
+        },
+        options: {
+            responsive: false,
+            legend: {
+                display: false,
+            },
+            scales: {
+                xAxes: [{
+                    scaleLabel: {
+                        display: true,
+                        labelString: 'Basis States'
+                    },
+                    ticks: {
+                        maxRotation: 0,
+                        minRotation: 0
+                    }
+                }],
+                yAxes: [{
+                    scaleLabel: {
+                        display: true,
+                        labelString: 'Meas. Probability'
+                    },
+                    ticks: {
+                        beginAtZero: true,
+                        suggestedMax: 1,
+                        suggestedMin: 0
+                    }
+                }]
+            }
+        }
+    });
+
+    //create legend and append to state div
+    let body = document.createElement("div");
+    body.id = "body";
+   
+    let state_div = state.div_id;
+    let div = document.getElementById(state_div);
+    div.appendChild(body);
+
+    //style the legend
+    document.getElementById("body").style.border = "thin solid #000000";
+    document.getElementById("body").style.textAlign = "center";
+
+    let legendTitle = document.createElement("p");
+    legendTitle.innerHTML = "Select Graph";
+
+
+    //create buttons
+    let button = document.createElement("button");
+    button.innerHTML = "amplitude^2";
+    button.id = "ye";
+
+    let button2 = document.createElement("button");
+    button2.innerHTML = "amplitude/phase";
+    button2.id = "ye";
+
+    var button3 = document.createElement("button");
+    button3.innerHTML = "real/imag";
+    button3.id = "ye";
+
+    //append button to legend
+    body.appendChild(legendTitle);
+    body.appendChild(button);
+    body.appendChild(button2);
+    body.appendChild(button3);
+
+    //create event listeners
+    button.addEventListener("click", function() {
+        //step one: remove each item currently in the dataset
+        let i = 0;
+        for (i = 0; i < newCount; i++){
+            removeData(measurementHistogram);
+        }
+
+        for (i = 0; i < newCount; i++) {
+            removeData(measurementHistogram);
+        }
+        //step two: repopulate the canvas with the new data
+        let y = Array.from(Array(newCount).keys()).map(idx => {
+            let bitstring = (idx >>> 0).toString(2).padStart(newCount, "0");
+            return `|${bitstring}⟩`;
+        });
+        let x = Array.from(Array(newCount).keys()).map(idx => {
+            return (amps[idx].Magnitude ** 2);
+        });
+
+        for (i = 0; i < newCount; i++){
+            addData(measurementHistogram, y[i], x[i]);
```

Comment:
Maybe I'm mistaken what `newCount` is, but should `addData` be affecting it? i.e. when we perform `addData`, should `newCount` be updated so that the next click event would remove those extra `newCount`s in L140, for example?